core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -407,36 +408,38 @@ public static void configureHadoopConf(Object maybeConfigurable, Object conf) {
    *
    * <p>The implementation must have a no-arg constructor.
    *
-   * @param impl full class name of a custom {@link MetricsReporter} implementation
+   * @param properties catalog properties which contains class name of a custom {@link
+   *     MetricsReporter} implementation
    * @return An initialized {@link MetricsReporter}.
-   * @throws IllegalArgumentException if class path not found or right constructor not found or the
-   *     loaded class cannot be cast to the given interface type
    */
-  public static MetricsReporter loadMetricsReporter(String impl) {
+  public static MetricsReporter loadMetricsReporter(Map<String, String> properties) {
+    String impl = properties.get(CatalogProperties.METRICS_REPORTER_IMPL);
+    if (impl == null || impl.isEmpty()) {
+      return LoggingMetricsReporter.instance();
+    }
     LOG.info("Loading custom MetricsReporter implementation: {}", impl);
     DynConstructors.Ctor<MetricsReporter> ctor;
+    MetricsReporter reporter = null;
     try {
       ctor =
           DynConstructors.builder(MetricsReporter.class)
               .loader(CatalogUtil.class.getClassLoader())
               .impl(impl)
               .buildChecked();
+      reporter = ctor.newInstance();
     } catch (NoSuchMethodException e) {
-      throw new IllegalArgumentException(
-          String.format("Cannot initialize MetricsReporter, missing no-arg constructor: %s", impl),
+      LOG.warn(
+          "Cannot initialize MetricsReporter, missing no-arg constructor: {}, fall back to default LoggingMetricsReporter",
+          impl,
           e);
-    }
-
-    MetricsReporter reporter;
-    try {
-      reporter = ctor.newInstance();
     } catch (ClassCastException e) {
-      throw new IllegalArgumentException(
-          String.format(
-              "Cannot initialize MetricsReporter, %s does not implement MetricsReporter.", impl),
+      LOG.warn(
+          "Cannot initialize MetricsReporter, {} does not implement MetricsReporter, fall back to default LoggingMetricsReporter",
+          impl,
           e);
     }
-
+    reporter = reporter == null ? LoggingMetricsReporter.instance() : reporter;
+    reporter.initialize(properties);
```

Comment:
This should be the only addition in this method.

Also, please separate it from the surrounding by newlines since this is a separate task to prepare the reporter.

Reply:
done

Reply:
There should be an empty line after this to separate it from the return statement.

Reply:
done